File: SmartPhone++/SmartPhone.Model/Requests/CartItemUpsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Model.Requests
{
    public class CartItemUpsertRequest
    {
        [Required]
        [Range(1, int.MaxValue)]
        public int Quantity { get; set; }
        
        [Required]
        public int CartId { get; set; }
        
        [Required]
        public int ProductId { get; set; }
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/cart_item_upsert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'cart_item_upsert_request.g.dart';

@JsonSerializable()
class CartItemUpsertRequest {
  final int quantity;
  final int productId;
  final int cartId;

  CartItemUpsertRequest({
    required this.quantity,
    required this.productId,
    required this.cartId,
  });

  factory CartItemUpsertRequest.fromJson(Map<String, dynamic> json) => _$CartItemUpsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CartItemUpsertRequestToJson(this);
}
